server/roomlogs.ts

```diff
@@ -296,8 +296,8 @@ export class Roomlog {
 	}
 	async rename(newID: RoomID): Promise<true> {
 		await Rooms.Modlog.rename(this.roomid, newID);
-		this.roomid = newID;
 		if (roomlogTable) {
+			this.roomid = newID;
 			if (!(!Config.logchat || this.roomid.startsWith('battle-'))) {
 				await roomlogTable.updateAll({roomid: this.roomid})`WHERE roomid = ${this.roomid}`;
 			}
```

Comment:
```suggestion
			if (!(!Config.logchat || this.roomid.startsWith('battle-'))) {
				await roomlogTable.updateAll({roomid: newID})`WHERE roomid = ${this.roomid}`;
			}
			this.roomid = newID;
```

Reply:
Also ```ts
			Roomlogs.roomlogs.delete(this.roomid);
		```
This should be somewhere before the main `if`.

Reply:
If this.destroy() gets moved to the top, wouldn't that make `roomlogStreamExisted` always be false?

Reply:
I moved that variable to before the `this.destroy()` call